torchbenchmark/models/nvidia_deeprecommender/nvinfer.py

```diff
@@ -0,0 +1,239 @@
+
+    if usecommandlineargs:
+      self.args = getCommandLineArgs()
+    else:
+      self.args = getBenchmarkArgs(forcecuda)
+
+    args = processArgState(self.args)
+
+    self.params = dict()
+    self.params['batch_size'] = 1
+    self.params['data_dir'] =  self.args.path_to_train_data
+    self.params['major'] = 'users'
+    self.params['itemIdInd'] = 1
+    self.params['userIdInd'] = 0
+    if not self.args.silent:
+      print("Loading training data")
+    
+    self.data_layer = input_layer.UserItemRecDataProvider(params=self.params)
+    
+    if not self.args.silent:
+      print("Data loaded")
+      print("Total items found: {}".format(len(self.data_layer.data.keys())))
+      print("Vector dim: {}".format(self.data_layer.vector_dim))
+  
+      print("Loading eval data")
+
+    self.eval_params = copy.deepcopy(self.params)
+    # must set eval batch size to 1 to make sure no examples are missed
+    self.eval_params['batch_size'] = 1
+    self.eval_params['data_dir'] = self.args.path_to_eval_data
+    self.eval_data_layer = input_layer.UserItemRecDataProvider(params=self.eval_params,
+                                                               user_id_map=self.data_layer.userIdMap,
+                                                               item_id_map=self.data_layer.itemIdMap)
+  
+    self.rencoder = model.AutoEncoder(layer_sizes=[self.data_layer.vector_dim] + [int(l) for l in self.args.hidden_layers.split(',')],
+                                      nl_type=self.args.non_linearity_type,
+                                      is_constrained=self.args.constrained,
+                                      dp_drop_prob=self.args.drop_prob,
+                                      last_layer_activations=not self.args.skip_last_layer_nl)
+  
+    self.path_to_model = Path(self.args.save_path)
+    if self.path_to_model.is_file():
+      print("Loading model from: {}".format(self.path_to_model))
+      self.rencoder.load_state_dict(torch.load(self.args.save_path))
+  
+    if not self.args.silent:
+      print('######################################################')
+      print('######################################################')
+      print('############# AutoEncoder Model: #####################')
+      print(self.rencoder)
+      print('######################################################')
+      print('######################################################')
+
+    self.rencoder.eval()
+  
+    if self.args.use_cuda: self.rencoder = self.rencoder.cuda()
+  
+    if self.args.jit:
+      self.rencoder = torch.jit.script(self.rencoder)
+  
+    self.inv_userIdMap = {v: k for k, v in self.data_layer.userIdMap.items()}
+    self.inv_itemIdMap = {v: k for k, v in self.data_layer.itemIdMap.items()}
+  
+    self.eval_data_layer.src_data = self.data_layer.data
+
+  def eval(self, niter=1):
+      for i, ((out, src), majorInd) in enumerate(self.eval_data_layer.iterate_one_epoch_eval(for_inf=True)):
```

Comment:
i didn't run the code, but i'm guessing the iterate_one_epoch_Eval function here is producing a whole sequence of training steps.

in our paradigm, we just want to take one step, and then we have benchmark driver logic calling your eval() function a bunch of times to takes statistical measurements - so we don't need or benefit from the averaging that goes on by running in a loop on the inside.  Instead, the benchmark driver will run this a minimum number of times, accruing a multiple of the slow 10s of sec runtime of this eval func.

Really, what you want to do in your wrapper is directly interact with the module.  in this case, the module is DeepRecommenderInferenceBenchmark.rencoder

Some folks take the approach of duplicating the setup code (code that builds rencoder) from this file in their own __init__ file.  It would also be fine and probably better if you just construct the DeepRecommenderInferenceBenchmark as you do, then reach into it and use the rencoder attribute directly in your wrapper.

Reply:
Can you clarify what do you mean by "a whole sequence of training steps"? First, the `iterate_one_epoch_eval` function is to run one epoch in eval, not train. For example, do you mean we should only run one iteration of the loop here(https://github.com/pytorch/benchmark/pull/373/files#diff-24c21d850287bf146bc126be1a92b29f448b6f4d55f621bd91defa367dc9a35cR111)?